compiler/src/dotty/tools/dotc/quoted/PickledQuotes.scala

```diff
@@ -241,7 +241,9 @@ object PickledQuotes {
         treeOwner(tree) match
           case Some(owner) =>
             // Copy the cached tree to make sure the all definitions are unique.
-            TreeTypeMap(oldOwners = List(owner), newOwners = List(owner)).apply(tree)
+            val treeCpy = TreeTypeMap(oldOwners = List(owner), newOwners = List(owner)).apply(tree)
+            // Then replace the symbol owner with the one pointed by the quote context.
+            treeCpy.changeNonLocalOwners(ctx.owner)
```

Comment:
I was curious why not propose this change, I've tried it locally with the test for #20471 but the following test (and only one) `tests/run-staging/quote-ackermann-1.scala` fails.
```suggestion
            TreeTypeMap(oldOwners = List(owner), newOwners = List(ctx.owner)).apply(tree)
            // Then replace the symbol owner with the one pointed by the quote context.
            //treeCpy.changeNonLocalOwners(ctx.owner)
```